builtin/providers/newrelic/resource_newrelic_alert_condition.go

```diff
@@ -0,0 +1,338 @@
+							ValidateFunc: validation.StringInSlice([]string{"all", "any"}, false),
+						},
+					},
+				},
+				Required: true,
+				MinItems: 1,
+			},
+			"user_defined_metric": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"user_defined_value_function": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"average", "min", "max", "total", "sample_size"}, false),
+			},
+		},
+	}
+}
+
+func buildAlertConditionStruct(d *schema.ResourceData) *newrelic.AlertCondition {
+	entitySet := d.Get("entities").([]interface{})
+	entities := make([]string, len(entitySet))
+
+	for i, entity := range entitySet {
+		entities[i] = strconv.Itoa(entity.(int))
+	}
+
+	termSet := d.Get("term").([]interface{})
+	terms := make([]newrelic.AlertConditionTerm, len(termSet))
+
+	for i, termI := range termSet {
+		termM := termI.(map[string]interface{})
+
+		terms[i] = newrelic.AlertConditionTerm{
+			Duration:     termM["duration"].(int),
+			Operator:     termM["operator"].(string),
+			Priority:     termM["priority"].(string),
+			Threshold:    termM["threshold"].(float64),
+			TimeFunction: termM["time_function"].(string),
+		}
+	}
+
+	condition := newrelic.AlertCondition{
+		Type:     d.Get("type").(string),
+		Name:     d.Get("name").(string),
+		Enabled:  true,
+		Entities: entities,
+		Metric:   d.Get("metric").(string),
+		Terms:    terms,
+		PolicyID: d.Get("policy_id").(int),
+	}
+
+	if attr, ok := d.GetOk("runbook_url"); ok {
+		condition.RunbookURL = attr.(string)
+	}
+
+	if attrM, ok := d.GetOk("user_defined_metric"); ok {
+		if attrVF, ok := d.GetOk("user_defined_value_function"); ok {
+			condition.UserDefined = newrelic.AlertConditionUserDefined{
+				Metric:        attrM.(string),
+				ValueFunction: attrVF.(string),
+			}
+		}
+	}
+
+	return &condition
+}
+
+func readAlertConditionStruct(condition *newrelic.AlertCondition, d *schema.ResourceData) error {
+	ids, err := parseIDs(d.Id(), 2)
+	if err != nil {
+		return err
+	}
+
+	policyID := ids[0]
+
+	entities := make([]int, len(condition.Entities))
+	for i, entity := range condition.Entities {
+		v, err := strconv.ParseInt(entity, 10, 32)
+		if err != nil {
+			return err
+		}
+		entities[i] = int(v)
+	}
+
+	d.Set("policy_id", policyID)
+	d.Set("name", condition.Name)
+	d.Set("type", condition.Type)
+	d.Set("metric", condition.Metric)
+	d.Set("runbook_url", condition.RunbookURL)
+	d.Set("user_defined_metric", condition.UserDefined.Metric)
+	d.Set("user_defined_value_function", condition.UserDefined.ValueFunction)
+	d.Set("entities", entities)
+
+	var terms []map[string]interface{}
+
+	for _, src := range condition.Terms {
+		dst := map[string]interface{}{
+			"duration":      src.Duration,
+			"operator":      src.Operator,
+			"priority":      src.Priority,
+			"threshold":     src.Threshold,
+			"time_function": src.TimeFunction,
+		}
+		terms = append(terms, dst)
+	}
+
+	d.Set("term", terms)
```

Comment:
We should maybe log an error here if we cannot set terms to state (as described previously)